models.py

```diff
@@ -595,18 +595,43 @@ def apply_dither(x: shaping.BT, dither: float):
 	else:
 		return x
 
+
 def entropy(log_probs, lengths = None, dim = 1, eps = 1e-9, sum = True, keepdim = False):
+	""" Calculate the entropy of probabilities, using log of probabilities, and applying the formula:
+			Entropy(p) = -SUM(pj * log pj).
+	"""
 	e = -(log_probs.exp() * log_probs).sum(dim = dim, keepdim = keepdim)
 	if lengths is not None:
 		e = e * temporal_mask(e, lengths)
-	return (
-		e.sum(dim = -1) / (eps + lengths.type_as(log_probs)) if lengths is not None else e.mean(dim = -1)
-	) if sum else e
+	if not sum:
+		return e
+	elif lengths is None:
+		return e.mean(dim = -1)
+	else:
+		return e.sum(dim = -1) / (eps + lengths.type_as(log_probs))
+
+
+def weighted_mean_entropy(log_probs, lengths = None, dim = -2, eps = 1e-9, epsilon_ind = -1):
```

Comment:
Please rename into `eps_id` to match existing naming in https://github.com/vadimkantorov/convasr/blob/9bab51ff907a95f39b107ac360c27e169237157a/text_tokenizers.py#L21

Reply:
done